lib/msal-browser/src/interaction_client/NativeInteractionClient.ts

```diff
@@ -48,8 +51,19 @@ export class NativeInteractionClient extends BaseInteractionClient {
 
         const reqTimestamp = TimeUtils.nowSeconds();
         const response: object = await this.nativeMessageHandler.sendMessage(messageBody);
-        const validatedResponse: NativeResponse = this.validateNativeResponse(response);
-        return this.handleNativeResponse(validatedResponse, nativeRequest, reqTimestamp);
+        const validatedResponse: NativeResponse = this.validateNativeResponse(response, nativeATMeasurement);
+
+        // generate the authentication result
+        const authResult = await this.handleNativeResponse(validatedResponse, nativeRequest, reqTimestamp, nativeATMeasurement);
```

Comment:
Don't pass `nativeATMeasurement` in here. If you want to measure `handeNativeResponse` on its own, start a new measurement in that function.

Reply:
It looks like you are doing that beacuse `handleNativeresponse` may throw, in which case here you should catch the error, end the measurement, and then rethrow the error.

Reply:
Or, move this measurement inside `handleNativeResponse` completely.

Reply:
Handled with try/catch